src/lib/cfa/sudoers.rb

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/target_file"
+require "cfa/base_model"
+require "cfa/augeas_parser"
+require "cfa/matcher"
+require "cfa/placer"
+
+module CFA
+  # Model to handle users specifications in the /etc/sudoers file
+  #
+  # @example Loading specifications
+  #   file = Sudoers.new
+  #   file.load
+  #   file.current_user_specs
+  #
+  # @example Shortcut loading specificitions
+  #   Sudoers.load.users_specs
+  #
+  # @example Writing new specifications
+  #   file = Sudores.new
+  #   file.load
+  #   file.clean_users
+  #   file.add_user({ "user" => "yast", "host" => "ALL", "commands" => ["/sbin/yast2"] })
+  #   file.add_user({ "user" => "foo", "host" => "ALL", "commands" => ["ALL"], "tag" => "NOPASSWD" })
+  #   file.save
+  class Sudoers < BaseModel
+    # @return [AgueasParser] the Augeas parser for sudoers file
+    PARSER = AugeasParser.new("sudoers.lns")
```

Comment:
well, I start to discourage to use constant parser that is shared between instances of Sudoers. It cause a bug in bootloader where problem was that there was multiple models instantiated and parser contain its file content cache, so it was also shared and result in some bugs. For more details see https://github.com/config-files-api/config_files_api_grub2/pull/19/files and https://bugzilla.suse.com/show_bug.cgi?id=1044409
So even if you know that now only one instance will be used, it is good to not share parser between them.